react/features/base/config/middleware.js

```diff
@@ -93,19 +93,33 @@ function _appWillMount(store, next, action) {
  * @private
  * @returns {*} The return value of {@code next(action)}.
  */
-function _setConfig({ getState }, next, action) {
+function _setConfig({ dispatch, getState }, next, action) {
     // The reducer is doing some alterations to the config passed in the action,
     // so make sure it's the final state by waiting for the action to be
     // reduced.
     const result = next(action);
+    const state = getState();
+
+    // Update the config with user defined settings.
+    const settings = state['features/base/settings'];
```

Comment:
I like how you made this update config action private and I understand that you're doing it here, because it's LJM that picks up the p2p enabled config value directly. I am only worried that having this code here could encourage others to put more and more stuff into the config, making /base/config depend on other features leading to circular dependencies issue. At the same time I like how you did the call integration, where it is the call integration feature that polls all the sources like settings and feature flags. So I am wondering if it's better to modify the config object passed to LJM in the place where it is initialized and poll all config sources like the call integration does? This way we would also not loose the original config value should it be needed for anything.

Reply:
yeah, I only hope it will be either me or you reviewing it...